apps/site/assets/ts/projects/components/MoreProjectsTable.tsx

```diff
@@ -7,41 +7,94 @@ interface Props {
   placeholderImageUrl: string;
 }
 
+interface State {
+  projects: Project[];
+  fetchInProgress: boolean;
+}
+
+export const fetchMoreProjects = (
```

Comment:
a few ideas for this function:

- return early (instead of wrapping in a big if statement)
- use spread operator when changing one value in state object
- async/await instead of promise (maybe this is just personal preference, but I find it easier to read)

```export const fetchMoreProjects = async (
  state: State,
  setState: Dispatch<State>
): Promise<void> => {
  if (!window.fetch) {
    return;
  }
  setState({ ...state, fetchInProgress: true });

  const offset = state.projects.length;
  const response = await window.fetch(`/project_api?offset=${offset}`);
  const projects: Project[] =
    response.ok && response.body ? Array.from(await response.json()) : [];

  setState({
    projects: state.projects.concat(projects),
    fetchInProgress: false
  });
};```